Arch Linux update from 0.4.7 to 0.5.1: panic: runtime error: invalid memory address or nil pointer dereference

Hello,
the Arch Linux package was finally updated to 0.5.1 today, so I went and updated my public node and personal node, and in both cases I receive the same issue:

upon launching yggdrasil with the new version using the old config file, I get the following error:

```bash
$ sudo yggdrasil -useconffile /etc/yggdrasil.conf
2023/11/03 12:47:16 Build name: yggdrasil
2023/11/03 12:47:16 Build version: 0.5.1
2023/11/03 12:47:16 Failed to connect to [2001:19f0:5:2c95:5400:1ff:fe46:9c]:34660: dial tcp [2001:19f0:5:2c95:5400:1ff:fe46:9c]:34660: connect: network is unreachable
2023/11/03 12:47:16 Failed to connect to [2001:19f0:5:6ed:5400:ff:fe5f:94f7]:12345: dial tcp [2001:19f0:5:6ed:5400:ff:fe5f:94f7]:12345: connect: network is unreachable
2023/11/03 12:47:16 Failed to connect to [2001:19f0:5:2a83:5400:1ff:fe89:44db]:58149: dial tcp [2001:19f0:5:2a83:5400:1ff:fe89:44db]:58149: connect: network is unreachable
2023/11/03 12:47:16 Failed to connect to [2604:a880:cad:d0::45:d001]:59168: dial tcp [2604:a880:cad:d0::45:d001]:59168: connect: network is unreachable
2023/11/03 12:47:16 Failed to connect to [2604:180:1:568::beef]:12345: dial tcp [2604:180:1:568::beef]:12345: connect: network is unreachable
2023/11/03 12:47:16 Failed to connect to [2600:3c00::f03c:91ff:feae:3efa]:44478: dial tcp [2600:3c00::f03c:91ff:feae:3efa]:44478: connect: network is unreachable
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x55b7f026cee4]

goroutine 56 [running]:
github.com/yggdrasil-network/yggdrasil-go/src/core.(*linkConn).SetDeadline(0xc14966e696d420e2?, {0x165ce249e?, 0x55b7f06c4b60?, 0x55b7f06c4b60?})
    <autogenerated>:1 +0x24
github.com/yggdrasil-network/yggdrasil-go/src/core.(*links).handler(0xc0000021e8, 0x0, {0x0, 0x0, {0x0, 0x0}, {0x0, 0x0, 0x0}}, {0x55b7f042e5e8, ...})
    github.com/yggdrasil-network/yggdrasil-go/src/core/link.go:537 +0x18c
github.com/yggdrasil-network/yggdrasil-go/src/core.(*links).add.func1.2()
    github.com/yggdrasil-network/yggdrasil-go/src/core/link.go:309 +0x4f9
created by github.com/yggdrasil-network/yggdrasil-go/src/core.(*links).add.func1 in goroutine 55
    github.com/yggdrasil-network/yggdrasil-go/src/core/link.go:240 +0x809
```

I've tried using the -normaliseconf option from the configuration documentation to update it (by replacing the old one with the normalized one), and the error doesn't leave, however, everything seems to be fine if I start it with autoconf, which makes it useless as you can't have any peers set up from the start in that case. I'm not sure what else to try to fix it. Any help would be appreciated.

Thank you.

--- a/src/core/link.go
+++ b/src/core/link.go
@@ -274,6 +274,9 @@ func (l *links) add(u *url.URL, sintf string, linkType linkType) error {
 
 				conn, err := l.connect(state.ctx, u, info, options)
 				if err != nil || conn == nil {
+					if err == nil && conn == nil {
+						l.core.log.Warnf("Link %q reached inconsistent error state", u.String())
+					}
 					if linkType == linkTypePersistent {
 						// If the link is a persistent configured peering,
 						// store information about the connection error so

--- a/src/core/link.go
+++ b/src/core/link.go
@@ -273,7 +273,7 @@ func (l *links) add(u *url.URL, sintf string, linkType linkType) error {
 				}
 
 				conn, err := l.connect(state.ctx, u, info, options)
-				if err != nil {
+				if err != nil || conn == nil {
 					if linkType == linkTypePersistent {
 						// If the link is a persistent configured peering,
 						// store information about the connection error so

